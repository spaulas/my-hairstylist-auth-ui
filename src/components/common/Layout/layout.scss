@import "@styles/imports";

.layout {
  position: relative;
  width: 100%;
  height: 100vh;
  max-width: 1440px;
  margin: 0 auto;

  &--strip {
    position: absolute;
    background-color: $primary--blue;
    width: 480px;
    top: 0;
    bottom: 0;
    right: calc(50% + 18px);
  }

  &--box {
    @include shadow--gray-medium;
    position: absolute;
    display: flex;
    flex-direction: row;
    background-color: $background--gray-light;
    width: 1068px;
    height: 680px;
    right: 0;
    left: 0;
    top: 0;
    bottom: 0;
    margin: auto;
    border-radius: 8px;

    & > * {
      width: 50%;
    }

    &-strip {
      margin: 36px 18px 36px 36px;
      background-color: $primary--blue;
    }

    &-children {
      margin: 36px 36px 36px 18px;
    }
  }
}
