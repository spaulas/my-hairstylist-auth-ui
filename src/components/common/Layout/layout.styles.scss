@import "@styles/imports";

.layout {
  position: relative;
  width: 100%;
  height: 100%;
  max-width: 1440px;
  margin: 0 auto;

  &--strip {
    position: absolute;
    background-color: $primary--blue;
    width: 480px;
    top: 0;
    bottom: 0;
    right: calc(50% + 18px);
  }

  &--box {
    @include shadow--gray__medium;
    position: absolute;
    display: flex;
    flex-direction: row;
    background-color: $background--gray__light;
    width: 1068px;
    height: 680px;
    right: 0;
    left: 0;
    top: 0;
    bottom: 0;
    margin: auto;
    border-radius: 8px;
    transition: all 0.4s ease;

    &__focused {
      @include shadow--gray__dark;
    }

    & > * {
      width: 50%;
    }

    &-strip {
      margin: 36px 18px 36px 36px;
      background-color: $primary--blue;
    }

    &-children {
      margin: 36px 36px 36px 18px;
    }
  }
}

@media only screen and (min-width: $tablet-landscape__min) and (max-width: $tablet-landscape__max) {
  .layout {
    max-width: 1140px;

    &--box {
      width: 840px;
      height: 534px;
    }

    &--strip {
      width: 366px;
    }
  }
}
