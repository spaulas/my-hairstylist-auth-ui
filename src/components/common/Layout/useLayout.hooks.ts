import { useSelector, useDispatch } from "react-redux";
import { RootReducerState } from "@type/global";
import layoutActions from "@actions/layout/layout.action";

const useLayout = () => {
  console.log("===================");
  const dispatch = useDispatch();

  const { isBackgroundFocused } = useSelector(
    ({ Layout }: RootReducerState) => {
      console.log("LAYOUT = ", Layout);
      return { isBackgroundFocused: Layout.isBackgroundFocused };
    }
  );
 

  useSelector((state) =>
    console.log("state - ", state)
  );
  const setBackgroundFocus = (focused: boolean) => {
    console.log("setBackgroundFocus focused = ", focused);
    dispatch(layoutActions.setBackgroundFocus(focused));
  };

  console.log("isBackgroundFocused = ", isBackgroundFocused);
  return { isBackgroundFocused, setBackgroundFocus };
};

export default useLayout;
