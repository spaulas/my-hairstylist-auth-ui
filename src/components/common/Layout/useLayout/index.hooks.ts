import { useSelector, useDispatch } from "react-redux";
import { RootReducerState } from "@type/global";
import layoutActions from "@actions/layout/layout.action";

const useLayout = () => {
  const dispatch = useDispatch();

  const { isBackgroundFocused } = useSelector(
    ({ Layout }: RootReducerState) => {
      return { isBackgroundFocused: Layout.isBackgroundFocused };
    }
  );

  const setBackgroundFocus = (focused: boolean) => {
    dispatch(layoutActions.setBackgroundFocus(focused));
  };

  return { isBackgroundFocused, setBackgroundFocus };
};

export default useLayout;
