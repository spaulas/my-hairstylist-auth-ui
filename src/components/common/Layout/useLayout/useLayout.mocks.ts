export const mockUseLayout = jest.fn();

jest.mock("./index.hooks.ts", () => mockUseLayout);
