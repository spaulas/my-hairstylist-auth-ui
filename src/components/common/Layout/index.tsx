import "./layout.styles.scss";
import useLayout from "./useLayout/index.hooks";

const Layout = ({ children }: any) => {
  const { isBackgroundFocused } = useLayout();

  return (
    <div className="layout">
      <div className="layout--strip" />
      <section
      data-testid="layoutBox"
        className={`layout--box ${
          isBackgroundFocused ? "layout--box__focused" : ""
        }`}
      >
        <div className="layout--box-strip" />
        <div className="layout--box-children">{children}</div>
      </section>
    </div>
  );
};

export default Layout;
