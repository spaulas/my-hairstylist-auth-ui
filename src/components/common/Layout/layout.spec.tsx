import "@testing-library/jest-dom/extend-expect";
import { mockUseLayout } from "./useLayout/useLayout.mocks";
import Layout from ".";
import { render, screen } from "@testing-library/react";

describe("Layout", () => {
  const hookValues = {
    isBackgroundFocused: true,
    setBackgroundFocus: jest.fn(),
  };

  const childrenText = "Layout Children";

  const renderLayout = () =>
    render(
      <Layout>
        <div>{childrenText}</div>
      </Layout>
    );

  beforeEach(() => {
    mockUseLayout.mockReturnValue(hookValues);
  });

  it("renders correctly", () => {
    const { container } = renderLayout();
    expect(container).toMatchSnapshot();
  });

  describe("layout--box should", () => {
    it("have layout--box__focused if isBackgroundFocused", () => {
      renderLayout();
      expect(screen.getByTestId("layoutBox")).toHaveClass(
        "layout--box__focused"
      );
    });

    it("not have layout--box__focused if !isBackgroundFocused", () => {
      const backgroundUnfocused = { ...hookValues, isBackgroundFocused: false };
      mockUseLayout.mockReturnValue(backgroundUnfocused);

      renderLayout();
      expect(screen.getByTestId("layoutBox")).not.toHaveClass(
        "layout--box__focused"
      );
    });
  });

  it("renders children components", () => {
    renderLayout();
    expect(screen.getByText(childrenText)).toBeTruthy();
  });
});
