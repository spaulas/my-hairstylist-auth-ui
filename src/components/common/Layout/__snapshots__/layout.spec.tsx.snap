// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="layout--strip"
    />
    <section
      class="layout--box layout--box__focused"
      data-testid="layoutBox"
    >
      <div
        class="layout--box-strip"
      />
      <div
        class="layout--box-children"
      >
        <div>
          Layout Children
        </div>
      </div>
    </section>
  </div>
</div>
`;
